src/libs/ajax/DataRepo.ts

```diff
@@ -142,11 +195,28 @@ export interface JobModel {
   class_name?: string;
 }
 
+export interface GetConceptsResponse {
+  result: SnapshotBuilderConcept[];
+}
+
+export interface GetConceptHierarchyResponse {
+  readonly result: SnapshotBuilderParentConcept[];
+}
+
+export interface SearchConceptsResponse {
+  result: SnapshotBuilderConcept[];
+}
+
+export interface SnapshotBuilderParentConcept {
+  parentId: number;
+  children: SnapshotBuilderConcept[];
+}
+
 export interface DataRepoContract {
   dataset: (datasetId: string) => {
     details: (include?: DatasetInclude[]) => Promise<DatasetModel>;
     roles: () => Promise<string[]>;
-    createSnapshotRequest(request: DatasetAccessRequest): Promise<DatasetAccessRequestApi>;
+    createSnapshotRequest(request: DatasetAccessRequestUtils): Promise<DatasetAccessRequest>;
```

Comment:
Why do we want to change this from the API type? Or is this just a renaming change?

Reply:
This is a renaming change

Reply:
The request is `DatasetAccessRequestUI` and it's promised the type `DatasetAccessRequest`. This should be `DatasetAccessResponse` but I believe from what I see in our tests, what's passed in is the same as the return.

Reply:
I'll add this to the refactoring ticket, this is a bit confusing. I like the convention of having `Response` at the end.